lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/ShelleySpec.hs

```diff
@@ -153,3 +200,21 @@ instance {-# OVERLAPS #-} Arbitrary (Passphrase "seed") where
         n <- choose (minSeedLengthBytes, 64)
         bytes <- BS.pack <$> vectorOf n arbitrary
         return $ Passphrase $ BA.convert bytes
+
+newtype SingleAddress = SingleAddress Address deriving (Eq, Show)
+
+instance Arbitrary SingleAddress where
+    arbitrary = SingleAddress . Address . BS.pack
+        <$> vectorOf addrSingleSize arbitrary
+
+newtype GroupedAddress = GroupedAddress Address deriving (Eq, Show)
+
+instance Arbitrary GroupedAddress where
+    arbitrary = GroupedAddress . Address . BS.pack
+        <$> vectorOf addrGroupedSize arbitrary
+
+newtype InvalidAddress = InvalidAddress Address deriving (Eq, Show)
+
+instance Arbitrary InvalidAddress where
+    arbitrary = InvalidAddress . Address . BS.pack
```

Comment:
maybe it would be good to be more diverse here. We want to check that 
1. `KeyFingerprint "payment" ShelleyKey` will get either `addrSingleSize` or `addrGroupedSize`. Otherwise error.
2. `KeyFingerprint "delegation" ShelleyKey` will get `addrGroupedSize`. when `addrSingleSize` it will return Nothing. Otherwise, error

Now we expose it with `addrSingleSize + addrGroupedSize` size address. Maybe we can be more diverse here ?